pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -103,14 +91,14 @@ func syncCluster(
 		}
 	}()
 
-	// Ignore stale poll-driven handler for a resource we have since
+	// Ignore stale poll-driven reconciler for a resource we have since
```

Comment:
As per my super-long review comment, this code comment will need to be changed a bit. And the ordering of this call with handleFinalizers below should probably be switched.

Reply:
I made the changes you recommended above and tested them. I didn't see any issues so I'm going to check them in.